c7n/resources/appmesh.py

```diff
@@ -27,23 +27,54 @@ class AppmeshMesh(QueryResourceManager):
 
     # interior class that defines the aws metadata for resource
     class resource_type(TypeInfo):
+        # service: is used by the boto client to look up the correct API for this resource.
         service = 'appmesh'
 
         # https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsappmesh.html#awsappmesh-resources-for-iam-policies   # noqa
+        # TODO: IS THIS EVEN USED AT RUNTIME?
+        # Junk value doesn't seem to affect the functionality of the extension
+        # but does fail the test PolicyMetaLint.test_valid_arn_type
+        # but that isn't a functional test.
         arn_type = "mesh"
 
         # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-virtualnode.html  # noqa
-        cfn_type = config_type = 'AWS::AppMesh::Mesh'
 
-        # Field in response containing the identifier used in API's.
-        # Therefore, this "id" field might be the arn field for some API's but
+        # TODO: IS THIS EVEN USED AT RUNTIME?
+        # Junk value doesn't seem to affect the functionality of the extension
+        # but does fail the test PolicyMetaLint.test_cfn_resource_validity
+        # but that isn't a functional test.
+        cfn_type = 'AWS::AppMesh::Mesh'
+
+        # TODO: IS THIS EVEN USED AT RUNTIME?
+        # Junk value doesn't seem to affect the functionality of the extension
+        # but does fail the test PolicyMetaLint.test_cfn_resource_validity
+        # but that isn't a functional test.
+        config_type = 'AWS::AppMesh::Mesh'
+
+        # id: Names the field in the response that contains the identifier to use
+        # in API calls to this service.
+        # Therefore, this "id" field might be the "arn" field for some API's but
         # in the case of Appmesh" it needs to be the field that contains the
         # name of the mesh as that's what the appmesh API's expect.
         # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-mesh.html   # noqa
-        id = name = 'meshName'
+        id = 'meshName'
 
-        # if a resource type is supported by resource group tagging
-        # api setting this value get tag filters/actions
+        # TODO: WHERE IS THIS EVEN USED AT RUNTIME?
+        name = 'meshName!!!!!!'

```

Comment:
This value isn't going to break functionality at runtime, it's more of a display thing. For example it shows up in [API stats recording](https://github.com/cloud-custodian/cloud-custodian/blob/503d73d89d3b97e94d97b2c8bdc164d0acac40e8/c7n/resources/aws.py#L685-L687) or in the [default field list of `custodian report`](https://github.com/cloud-custodian/cloud-custodian/blob/503d73d89d3b97e94d97b2c8bdc164d0acac40e8/c7n/reports/csvout.py#L161-L162).

Reply:
Is the "name" a field name in the resources, or is it a literal value?
Can't tell from the doco.

Reply:
It's a field name